### ISSUE TITLE
Support for Steam Deck "Dynamic Cloud Sync"

### ISSUE THREAD

**Issue by Belgerum** [renpy/renpy#3634]
Steam has a feature for "Dynamic Cloud Sync", letting players go back and forth between a Steam Deck and other PC, having cloud saves sync even if the game is not closed.

Steam's documentation about the feature:
https://partner.steamgames.com/doc/features/cloud#dynamic_cloud_sync

This is currently unsupported by Ren'Py. It would be nice to have this feature available for players, even if it's a little bit of a niche use currently.

**Comment by renpytom** [comment]
Unless I'm missing something, this should probably just work without dedicated support from Ren'Py.

**Comment by Belgerum** [comment]
I just enabled this for one of my steam games to test the feature, and it does work for general save data. When resuming a suspended game after the cloud saves have changed, any new or modified save slots appear and are loadable.

However, PERSISTENT data (such as the "preferences" menu options, language used in translations, dialogue and files marked as "seen", and more) does not update automatically. Steam cloud does seem to sync and update the persistent data file(s) to the correct directory on the Steam Deck, and Ren'Py does use the data from the file after closing or reloading the game, but it doesn't use it automatically when resuming a suspended session on the Deck, which never reloads the game or updates the persistent data from the cloud save like this feature would suggest it should.

Basically, to fully support this feature, Ren'Py needs to reload/update persistent data when it detects a Steam Deck is resuming a suspended session with changed/updated cloud save files.

**Comment by Belgerum** [comment]
Should I assume that this feature will remain unsupported for the foreseeable future?

After seeing this issue was closed, I tested Dynamic Sync with my Steam Deck using a build made with renpy-8.4.2.25090201+nightly-sdk

It still does not properly load any updated persistent data from the cloud to a suspended session that is resumed. Only any updated .save files from the cloud are recognized after the resume, and only after the save/load screen is shown again in-game.

**Comment by renpytom** [comment]
It should. How do you know it's not happening? 

Specifically, when a new persistent file is detected, Ren'Py will merge it with the existing persistent file. The precise details of this are customizable, and there does need to be some support for more complex merges.

**Comment by Belgerum** [comment]
I've just been using the Preference sliders / toggles in my game's settings screen, since those are the some of the easiest persistent values to quickly adjust and see if the change carries over.

I mostly follow the same steps as listed in Steam's documentation for "Pre-release Testing" at https://partner.steamgames.com/doc/features/cloud#dynamic_cloud_sync
except with a few differences, since I'm using a real Steam Deck to do the suspending/resuming instead of console commands.

Are you saying that when the persistent data is merged, some persistent data is deliberately kept? Should I be testing with other persistent data types like seen dialogue or user-defined variables instead?

EDIT: It may also be worth noting I have only been using the Install directory /game/saves/ for the Steam Cloud files for all platforms. My main PC runs windows and has the additional backup saves in /Users/Username/Appdata/, and don't know if that is affecting anything, nor if the Steam Deck running on Linux has its own backup location.

**Comment by renpytom** [comment]
I believe preferences are machine local, while other persistent data is merged - mostly because some of the preferences, like fullscreen mode, don't make sense on all platforms.
